### PR TITLE
nixos/mopidy: test & cleanup

### LINKED ISSUE #355993: nixos/mopidy: add package option
## Description

It would be great if it were possible to override the `mopidy` or `mopidyPackages` values in the service, i.e. to use packages from the unstable branch.

## Notify maintainers

No maintainers at this time apparently.

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Stunkymonkey**
adding test and cleanup

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/audio/mopidy.nix [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
 
-  mopidyConf = writeText "mopidy.conf" cfg.configuration;
+  mopidyConf = pkgs.writeText "mopidy.conf" cfg.configuration;
 
-  mopidyEnv = buildEnv {
-    name = "mopidy-with-extensions-${mopidy.version}";
-    paths = closePropagation cfg.extensionPackages;
-    pathsToLink = [ "/${mopidyPackages.python.sitePackages}" ];
-    nativeBuildInputs = [ makeWrapper ];
+  mopidyEnv = pkgs.buildEnv {
```
[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/services/audio/mopidy.nix:9]: good catch. merged it and will rebase

File: nixos/modules/services/audio/mopidy.nix [hunk 2]
```diff
@@ -1,23 +1,19 @@
...
+    name = "mopidy-with-extensions-${cfg.package.version}";
+    paths = lib.closePropagation cfg.extensionPackages;
+    pathsToLink = [ "/${pkgs.mopidyPackages.python.sitePackages}" ];
```
[the remark this answers is not in the document]
Comment (Stunkymonkey) [reply nixos/modules/services/audio/mopidy.nix:13]: should we provide an additional option for the `mopidyPackages`?

File: nixos/modules/services/audio/mopidy.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (jwillikers) [reply nixos/modules/services/audio/mopidy.nix:9]: Thanks!

File: nixos/modules/services/audio/mopidy.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (jwillikers) [reply nixos/modules/services/audio/mopidy.nix:13]: I was leaning toward something like that. It just makes me wonder if there still needs to be an option for `mopidy` then.

**Comment by Stunkymonkey** [comment]
@jwillikers maybe we first merge everything except the package option?

**Comment by jwillikers** [comment]
> @jwillikers maybe we first merge everything except the package option?

That sounds great to me!

**Review by jwillikers — APPROVED** [review]
Thanks for all the clean up!

**Pull request merged**